File: server/src/models/Tag.ts
import mongoose, { Schema } from 'mongoose';
import { v4 as uuidv4 } from 'uuid';

export interface ITag {
  _id: string;
  name: string;
  color: string;
  userEmail: string;
  createdAt: string;
}

const TagSchema = new Schema<ITag>({
  _id: { type: String, default: () => uuidv4() },
  name: { type: String, required: true },
  color: { type: String, default: '#58a6ff' },
  userEmail: { type: String, required: true },
  createdAt: { type: String, default: () => new Date().toISOString() },
});

TagSchema.index({ name: 1, userEmail: 1 }, { unique: true });

export const Tag = mongoose.model<ITag>('Tag', TagSchema);

File: server-java/src/main/java/com/conarum/tradingjournal/domain/tag/model/Tag.java
package com.conarum.tradingjournal.domain.tag.model;

import com.conarum.tradingjournal.common.model.BaseEntity;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@Document(collection = "tags")
@CompoundIndex(name = "name_userEmail_unique", def = "{'name': 1, 'userEmail': 1}", unique = true)
public class Tag extends BaseEntity {
    private String name;
    private String color = "#58a6ff";
}
